migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/V1SetterToV2.java

```diff
@@ -65,42 +75,44 @@ public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, Execu
                 selectType = select.getType();
             }
 
-            if (selectType == null) {
+            if (selectType == null || !shouldChangeSetter(selectType)) {
                 return method;
             }
 
-            if (shouldChangeSetter(method, selectType)) {
-                String methodName = method.getSimpleName();
+            String methodName = method.getSimpleName();
 
-                if (NamingUtils.isWither(methodName)) {
-                    methodName = NamingUtils.removeWith(methodName);
-                } else if (NamingUtils.isSetter(methodName)) {
-                    methodName = NamingUtils.removeSet(methodName);
-                }
+            if (NamingUtils.isWither(methodName)) {
+                methodName = NamingUtils.removeWith(methodName);
+            } else if (NamingUtils.isSetter(methodName)) {
+                methodName = NamingUtils.removeSet(methodName);
+            }
```

Comment:
Do you mean something like the following?
```
removeWithIfNeeded(methodName);
removeSetIfNeeded(methodName);
```

We just want to remove either `with` or `set` once though. It may be an issue for a field that has `set` or `with` in the name, something like `SetBucketLifecycle` (`withSetBucketLifeCyle`, `setSetBucketLifeCycle`)

Reply:
Yeah, that's what I meant, I missed that this was an else-if. Thanks!